File: 2753-count-houses-in-a-circular-street-ii/2753-count-houses-in-a-circular-street-ii.py
# Definition for a street.
# class Street:
#     def closeDoor(self):
#         pass
#     def isDoorOpen(self):
#         pass
#     def moveRight(self):
#         pass
class Solution(object):
    def houseCount(self, street, k):
        """
        :type street: Street
        :type k: int
        :rtype: int
        """
        while not street.isDoorOpen():
            street.moveRight()
        ans = 0
        for i in range(1, k + 1):
            street.moveRight()
            if street.isDoorOpen():
                ans = i
                street.closeDoor()
        return ans

File: 2753-count-houses-in-a-circular-street-ii/2753-count-houses-in-a-circular-street-ii.rb
# Definition for a street.
# class Street
#   def initialize(doors)
#   end
#   def close_door
#   end
#   def is_door_open
#   end
#   def move_right
#   end
# end
# @param {Street} street
# @param {Integer} k
# @return {Integer}
def house_count(street, k)
    until street.is_door_open
        street.move_right
    end
    ans = 0
    (1..k).each do |i|
        street.move_right
        if street.is_door_open
            ans = i
            street.close_door
        end
    end
    ans
end